### PR TITLE
feat(export-csv): add `Export CSV` button to download csv files

### PR DESCRIPTION
**Pull request by sahil143**
Story: https://openshift.io/openshiftio/Openshift_io/plan/detail/1247

**Backalog Tab**
screenshot from 2018-12-27 21-14-27

**Query Tab**
screenshot from 2018-12-27 21-14-34
screenshot from 2018-12-27 21-14-48

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @sahil143 for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin.

**Comment by christianvogt** [comment]
What happened to the formatting of the HTML?
Hard to track your changes.
Please ensure that you do not run any other formatter other than prettier.
What IDE do you use? If you use vscode, see the end of the `README.md` for extensions to install.

We should be enabling the full set of linting and formatting as soon as we can clean up the source to not have so many errors.

File: packages/planner/src/app/components_ngrx/planner-query/planner-query.component.less [hunk 1]
```diff
@@ -25,15 +25,21 @@
     top: 10px;
   }
 
+  &__export-csv {
```
Comment (christianvogt) [inline packages/planner/src/app/components_ngrx/planner-query/planner-query.component.less]: Avoid using absolute layouts unless you have a very good reason to do so. Always find a way to do a flow layout. Either use flex here or responsive grid classes.

**Comment by joshuawilson** [comment]
Please rebase.

**Comment by sahil143** [comment]
@joshuawilson @christianvogt I have rebased the branch and made the changes suggested by @Veethika  for `export-csv` button.
Added new screenshots in description.

File: packages/planner/src/app/components_ngrx/export-csv/export-csv.component.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+export class ExportCsvComponent {
+  @Input() isDisabled: boolean;
+  @Input() query: string;
+
+  public csvLink: Observable<string> = combineLatest(
+    this.spaceQuery.getCurrentSpace.pipe(filter((s) => !!s)),
+    this.route.queryParams.pipe(filter((q) => !!Object.keys(q).length)),
+  ).pipe(
+    switchMap(([space, query]) => {
+      console.log('################333333', query);
```
Comment (vikram-raj) [inline packages/planner/src/app/components_ngrx/export-csv/export-csv.component.ts]: Please remove the console log.

File: packages/planner/src/app/components_ngrx/export-csv/export-csv.component.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        this.baseApiURL +
+          'search/workitems/csv?' +
+          Object.keys(expression)
+            .map((k) => 'filter[' + k + ']=' + encodeURIComponent(JSON.stringify(expression[k])))
+            .join('&'),
+      );
+    }),
+  );
+
+  constructor(
```
Comment (joshuawilson) [inline packages/planner/src/app/components_ngrx/export-csv/export-csv.component.ts]: The style / standard we follow has the constructor listed above the functions.

File: packages/planner/src/app/components_ngrx/export-csv/export-csv.component.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import { Component, Input, Inject } from '@angular/core';
+import { FilterService } from '../../services/filter.service';
+import { WIT_API_URL, Space } from 'ngx-fabric8-wit';
```
Comment (joshuawilson) [inline packages/planner/src/app/components_ngrx/export-csv/export-csv.component.ts]: It appears that `space` isn't used.

**Comment by joshuawilson** [comment]
@sahil143 I don't see approval from @fabric8-ui/uxd on the location of the button. 
This also needs unit tests.

**Comment by Veethika** [comment]
Suggested to put the Export button on the top and move 'Show completed' and 'Show tree' to the bar below as the later is more connected to the table and should be placed closer
> 
> **Backalog Tab**
> screenshot from 2018-12-27 21-14-27
> 

Here, the suggestion was to move the export and create workitem button in the bar below, differentiated by color from the query input bar to avoid accidental click. Placing the button at the exact same position as 'backlog tab' wasn't possible with current design as it would involve adding another bar above the input field adding a lot more to the unused spaces. This could, however, be taken care in future design changes.
> **Query Tab**
> screenshot from 2018-12-27 21-14-34
> screenshot from 2018-12-27 21-14-48

The implemented changes LGTM!

**Comment by sahil143** [comment]
I'm adding unit tests for the component.

File: packages/planner/src/app/components_ngrx/export-csv/export-csv.component.ts (hunk 3 above)
Comment (sahil143) [reply packages/planner/src/app/components_ngrx/export-csv/export-csv.component.ts]: That's an Observable being assigned to `csvLink` variable. should I move it below constructor?

Comment (joshuawilson) [reply packages/planner/src/app/components_ngrx/export-csv/export-csv.component.ts]: I'm not blocking the merge on this, it just looks odd to me. It is not just a variable name that is being set. You are doing something there and that usually happens below the constructor. This is for readability not functionality.

Comment (sahil143) [reply packages/planner/src/app/components_ngrx/export-csv/export-csv.component.ts]: Done. Listed the `csvLink` below constructor.

**Pull request merged**